quiz/backend/stats/models.py

```diff
@@ -21,6 +21,29 @@ class Meta:
         abstract = True
 
 
+class PlayerBadge(models.Model):
+    """Model for linking badges :class:`stats.models.Badge` with player :class:`players.models.Player`
+
+    :param badge: ID of badge
+    :type badge: models.ForeignKey
+
+    :param player: ID of player
+    :type player: models.ForeignKey
+
+    :param obtained_on: Date when player obtained given badge.
+    :type obtained_on: models.DateTimeField
+
+    """
+
+    badge = models.ForeignKey("Badge", on_delete=models.DO_NOTHING)
+    player = models.ForeignKey("players.Player", on_delete=models.DO_NOTHING)
+    obtained_on = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        verbose_name = "badge obtained by player"
```

Comment:
Either both verbose with capital letter or both with lower letter.

Reply:
Don't know why I thought that verbose name plural is used only on the left side of the admin panel, changed it to lowercased 😅

Problem no. 1 please provide more information on how to reproduce this error, because no matter how hard I try I can access badges without any problems. 

Problem no. 2 rewritten validators and now file type should be checked first, looks like validators are run all at once, must investigate it.

Problem no. 3 reformated, must have forgotten at some point.

Reply:
resolved